deployment-scanner/src/main/java/org/jboss/as/server/deployment/scanner/FileSystemDeploymentService.java

```diff
@@ -931,6 +931,7 @@ private void scanDirectory(final File directory, final String relativePath, fina
                 final String deploymentName = fileName.substring(0, fileName.length() - FAILED_DEPLOY.length());
                 scanContext.toRemove.remove(deploymentName);
                 if (!deployed.containsKey(deploymentName) && !(new File(child.getParent(), deploymentName).exists())) {
+                    scanContext.scannerTasks.add(new UndeployTask(deploymentName, deploymentDir, child.lastModified(), true));
```

Comment:
I believe child.lastModified() should be scanContext.scanStartTime. If this actually does an undeploy, the undeploy didn't happen when the FAILED_DEPLOY marker was written.

It doesn't really matter because this param ends up unused since you pass 'true' as the forcedUndeploy param, but maybe after some future change it will matter.

Reply:
A more substantive problem is there needs to be a check that this isn't going to undeploy something not associated with this scanner; e.g. a scanned deployment failed, and admin removed the file, and then the admin used the CLI to force-deploy.

I believe scanContext.persistentDeployments is meant to contain the set of deployments not associated with this scanner, so that could be how to check.

Tests of this change should cover this scenario in addition to the basic one described in the JIRA.